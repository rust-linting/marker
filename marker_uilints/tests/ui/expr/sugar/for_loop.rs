#![feature(register_tool)]
#![register_tool(marker)]

fn main() {
    let range = 1..10;
    let mut total = 0;

    #[warn(marker::print_every_expr)]
    for i in range {
        total += i;
    }

    println!("{total}");
}
