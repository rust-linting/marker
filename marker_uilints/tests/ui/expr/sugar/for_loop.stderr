warning: expr
  --> $DIR/for_loop.rs:9:5
   |
9  | /     for i in range {
10 | |         total += i;
11 | |     }
   | |_____^
   |
   = note: SpanSource: File(
               FileInfo {
                   file: "$DIR/for_loop.rs",
                   span_src: SpanSrcId(..),
               },
           )
   = note: Snippet: "for i in range {/n        total += i;/n    }"
note: the lint level is defined here
  --> $DIR/for_loop.rs:8:12
   |
8  |     #[warn(marker::print_every_expr)]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^

warning: expr
 --> $DIR/for_loop.rs:9:14
  |
9 |     for i in range {
  |              ^^^^^
  |
  = note: SpanSource: File(
              FileInfo {
                  file: "$DIR/for_loop.rs",
                  span_src: SpanSrcId(..),
              },
          )
  = note: Snippet: "range"

warning: expr
  --> $DIR/for_loop.rs:9:20
   |
9  |       for i in range {
   |  ____________________^
10 | |         total += i;
11 | |     }
   | |_____^
   |
   = note: SpanSource: File(
               FileInfo {
                   file: "$DIR/for_loop.rs",
                   span_src: SpanSrcId(..),
               },
           )
   = note: Snippet: "{/n        total += i;/n    }"

warning: expr
  --> $DIR/for_loop.rs:10:9
   |
10 |         total += i;
   |         ^^^^^^^^^^
   |
   = note: SpanSource: File(
               FileInfo {
                   file: "$DIR/for_loop.rs",
                   span_src: SpanSrcId(..),
               },
           )
   = note: Snippet: "total += i"

warning: expr
  --> $DIR/for_loop.rs:10:18
   |
10 |         total += i;
   |                  ^
   |
   = note: SpanSource: File(
               FileInfo {
                   file: "$DIR/for_loop.rs",
                   span_src: SpanSrcId(..),
               },
           )
   = note: Snippet: "i"

warning: 5 warnings emitted

